bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/ConfigTomlParser.java

```diff
@@ -99,10 +101,6 @@ private static TomlTableNode retrieveModuleNode(TomlTableNode tomlNode, Module m
         }
         TomlTableNode moduleNode = moduleName.equals(DEFAULT_MODULE) ? tomlNode : extractModuleNode(tomlNode,
                 moduleName);
-        if (moduleNode == null) {
-            throw new TomlException(INVALID_TOML_FILE + "Module '" + moduleName + "' from organization '" + orgName
-                    + "' not found.");
-        }
```

Comment:
can we revert this change after we get information about the optional-ness of the config variables form the front-end ?

Reply:
yes. we should revert this.  And also the other checks we removed due to this

Reply:
I created this issue to track it #28018